src/core/filesystem.cc

```diff
@@ -1029,10 +1035,54 @@ S3FileSystem::DownloadFileFolder(
   return Status::Success;
 }
 
+int
+IsPathDirectory(const char* path)
+{
+  struct stat s_buf;
+
+  if (stat(path, &s_buf))
+    return 0;
+
+  return S_ISDIR(s_buf.st_mode);
+}
+
+int
+DeleteFolderRecursive(const std::string& path)
```

Comment:
Move these two functions to be `XXFileSystem` class members. And only implement in LocalFileSystem

Reply:
Fixed